map/chart_generator.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "indexer/feature_altitude.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "std/deque.hpp"
+#include "std/vector.hpp"
+
+/// \brief fills uniformAltitudeDataM with altitude data which evenly distributed by
+/// |resultPointCount| points. |distanceDataM| and |altitudeDataM| form a curve of route altitude.
+/// This method is used to generalize and evenly distribute points of the chart.
+void NormalizeChartData(deque<double> const & distanceDataM,
```

Comment:
So what? What's the problem to insert an element in a vector? Or, even better, why you can't do following:
```c++
vector<double> segDistanceM;
segDistanceM.push_back(0.0);
if (!m_routingSession.GetSegDistanceM(...) ...)
  ...
```